app/models/checklist_assignment.rb

```diff
@@ -26,7 +27,19 @@ class ChecklistAssignment < ApplicationRecord
   scope :completed, -> { where.not(completed_at: nil) }
   scope :incomplete, -> { where(completed_at: nil) }
 
+  scope :required, -> { joins(:checklist_template_item).where(checklist_template_items: {required: true}) }
+  scope :optional, -> { joins(:checklist_template_item).where(checklist_template_items: {required: false}) }
+
+  before_create :set_due_date
```

Comment:
I would move this `before_validation` with `on: :create` so that you can add a validation on-top of the `due_date` (I noticed we do not allow null on this column)